cli/azd/pkg/account/manager.go

```diff
@@ -0,0 +1,250 @@
+package account
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/azcli"
+	"golang.org/x/exp/slices"
+)
+
+// Manages azd account configuration
+// `az cli` is not required and will only be called `azd` default have not already been set.
+type Manager struct {
+	azCli azcli.AzCli
+}
+
+// Creates a new Account Manager instance
+func NewManager(ctx context.Context) *Manager {
+	return &Manager{
+		azCli: azcli.GetAzCli(ctx),
+	}
+}
+
+// Gets the default subscription for the logged in account.
+// 1. Returns AZD config defaults if exists
+// 2. Returns AZ CLI defaults if exists
+func (m *Manager) GetAccountDefaults(ctx context.Context) (*config.Config, error) {
```

Comment:
This should be returning a smaller `Account` struct, not the entire config object